tfe/resource_tfe_team.go

```diff
@@ -105,8 +105,13 @@ func resourceTFETeamCreate(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("[DEBUG] Create team %s for organization: %s", name, organization)
 	team, err := tfeClient.Teams.Create(ctx, organization, options)
 	if err != nil {
-		return fmt.Errorf(
-			"Error creating team %s for organization %s: %v", name, organization, err)
+		if err == tfe.ErrResourceNotFound {
+			entitlements, _ := tfeClient.Organizations.Entitlements(ctx, organization)
+			if !entitlements.Teams {
```

Comment:
a panic might happen here if `entitlements` is nil, so it may be best to check for the error return value here before dereferencing entitlements

It may be interesting to eventually pull entitlements up front so that all resources could short circuit with a helpful error message some day. I didn't know about this API. I'll look into it!

Reply:
It could be something as simple as the API returns a `500` or a `503`. If an organization has zero entitlements, that would be a huge problem and beyond the scope of the provider.

I don't think an explicit test is necessary; simply having a block that handles the error and returns a clear error msg is sufficient.

Reply:
I would just fall through and let the 404 error be returned

Reply:
gotcha! that makes a lot of sense. thanks for clarifying that 😊. Just pushed changes for this feedback